src/compile/scale/parse.ts

```diff
@@ -75,10 +75,11 @@ function parseUnitScaleCore(model: UnitModel): ScaleComponentIndex {
       specifiedScale = specifiedScale || {};
       const specifiedScaleType = specifiedScale.type;
       const sType = scaleType(specifiedScale.type, channel, fieldDef, mark, config.scale);
-      scaleComponents[channel] = new ScaleComponent(
+      const component = new ScaleComponent(
```

Comment:
Not sure why this has to be changed here.

Reply:
Done